cmd/kube-apiserver/app/server.go

```diff
@@ -244,14 +245,22 @@ func CreateNodeDialer(s completedServerRunOptions) (tunneler.Tunneler, *http.Tra
 		// Get ssh key distribution func, if supported
 		var installSSHKey tunneler.InstallSSHKey
 
-		cloudprovider.DeprecationWarningForProvider(s.CloudProvider.CloudProvider)
```

Comment:
seems like a nice signal just to let users know it will be going away with the full conversion to out-of-tree. that said, i don't have deep experience on the warning we should be emitting in this part of the code.

Reply:
This deprecation warning is only displaying something for deprecated in-tree cloud provider selections. If the feature gate is set the code won't reach there, but until the FeatueGate is in alpha and is allowed to be disabled we should keep it in place.

Reply:
I added a similar block in IsDeprecatedInternal.